packages/functions/src/speaking.ts

```diff
@@ -3,119 +3,173 @@ import {
+
+  return {
+    statusCode: 200,
+    body: JSON.stringify(output),
+  };
+};
+
+/*
+  This function starts a transcription job and returns only when that
+  job is 'COMPLETE', the while loop is necessary because without it the 
+  function will return when the transcription job is submitted successfully.
+*/
+async function startTranscription(audioFileName: string) {
   try {
-    const commandOutput = await s3Client.send(
-      new GetObjectCommand({
-        Bucket: transcribeBucketName,
-        Key: key,
+    const transcript = await transcribeClient.send(
+      new StartTranscriptionJobCommand({
+        TranscriptionJobName: audioFileName,
+        LanguageCode: 'en-US',
+        MediaFormat: 'webm',
+        Media: {
+          MediaFileUri: `s3://${uploadResponseBucket}/${audioFileName}.webm`,
+        },
+        OutputBucketName: uploadResponseBucket,
       }),
     );
-    if (!commandOutput.Body) {
-      return;
-    }
-    const content = await commandOutput.Body.transformToString('utf-8');
-    if (content.length === 0) {
-      console.log(`file ${key} is empty`);
-    } else {
-      const answer = JSON.parse(content).results.transcripts[0].transcript;
-      const prompt = createPrompt(rubric, question, answer);
-      const input = {
-        inputText: prompt,
-        textGenerationConfig: {
-          maxTokenCount: 4096,
-          stopSequences: [],
-          temperature: 0,
-          topP: 0.9,
-        },
-      };
-
-      const command = new InvokeModelCommand({
-        body: JSON.stringify(input),
-        contentType: 'application/json',
-        accept: '*/*',
-        modelId: 'amazon.titan-text-express-v1',
-      });
-
-      const response = await bedrockClient.send(command);
-      const response_byte = response.body;
-
-      const textDecoder = new TextDecoder('utf-8');
-      const decodedString = textDecoder.decode(response_byte);
-
-      const feedbackResult = JSON.parse(decodedString).results[0].outputText;
-
-      const score_index = feedbackResult.indexOf('Score:');
-      const feedback_index = feedbackResult.indexOf('Feedback:');
-      const score = feedbackResult
-        .substring(score_index + 'Score: '.length, feedback_index)
-        .trim();
-      const feedback = feedbackResult
-        .substring(feedback_index + 'Feedback: '.length)
-        .trim();
-      console.log(score);
-      console.log(feedback);
-
-      const item = {
-        feedbackId: { S: `${Math.random()}` },
-        feedbackScore: { S: score },
-        feedbackText: { S: feedback },
-      };
-      try {
-        const command = new PutItemCommand({
-          TableName: FeedbackTableName,
-          Item: item,
-        });
-        await dynamoClient.send(command);
-        return;
-      } catch (error) {
-        console.log('Error storing result');
-        return;
-      }
+    let status = transcript.TranscriptionJob?.TranscriptionJobStatus;
+    while (status != 'COMPLETED') {
```

Comment:
Keep in mind that the transcription job may fail (`status === 'FAILED'`).  It seems like your code will have an infinite loop in that case.

You should handle failure properly.

Reply:
Good catch.
Fixed in #[da3147c](https://github.com/bahrain-uob/gen-ai-moe-challenge/pull/83/commits/da3147cb574b3e330eb8b492bad9933087162418)